{
  "title": "IDApython confused with get_32bit(ea) function behavior",
  "link": "https://reverseengineering.stackexchange.com/questions/22384/idapython-confused-with-get-32bitea-function-behavior",
  "content": "I am using some code like this \n\n```\n   import idaapi\n   print(idaapi.get_32bit(0x0055f4a0))\n\n```\nit must return 32 bit of address as int\n\nbut it returns 1408011093\nis it wrong conversion?\nits length is less than the maximum length of int \n\nbut it somehow changes the value \nI expected to get return value like ‭5633184‬\n\n",
  "votes": "1",
  "answers": 1,
  "views": "156",
  "tags": [
    "idapython"
  ],
  "user": "Huntermike",
  "time": "Oct 24, 2019 at 18:03",
  "comments": [],
  "answers_data": [
    {
      "content": "get_32bit(0x0055f4a0) returns the 32-bit value in your IDB at the address 0x0055f4a0. 1408011093 decimal is 0x53EC8B55 in hex which fits in 32 bits. In little endian format this corresponds to the byte sequence 55 8B EC 53, so 0x0055f4a0 is likely a function start.\n\nTo convert hex to decimal you don't need IDA APIs but just standard Python functionality, e.g. str(int(0x0055f4a0)). But this is best discussed on Stack Overflow since that's a programming question and not RE.\n\n",
      "votes": "0",
      "user": "Igor Skochinsky",
      "time": "Oct 24, 2019 at 19:11",
      "is_accepted": true,
      "comments": [
        {
          "user": "Huntermike",
          "text": "ty so  just need to go through the standard conversion",
          "time": null
        }
      ]
    }
  ]
}